eZ/Publish/Core/Repository/ContentService.php

```diff
@@ -1826,6 +1827,10 @@ protected function internalPublishVersion(APIVersionInfo $versionInfo, $publicat
             100 // Limited to avoid publishing taking to long, besides SE limitations this is why limit is max 50
         );
 
+        if (!$this->settings['remove_archived_versions_on_publish']) {
+            return $content;
+        }
+
```

Comment:
I'm not ok with early return in this case. I'd prefer it to move the archivization part of this method to a separate method and wrap it with a logical condition (as it's a specific operation we want to skip).

I hope that won't lead into conflicts during merge up, but I think that's a proper approach here.

Reply:
Yup, totally agree, wanted to change it to this morning, didnt expect review at 10PM ;)